### ISSUE TITLE
`code` executed in Git Bash terminal does not refer to the local VSCode

### ISSUE THREAD

**Issue by pawelszramowski** [microsoft/vscode-remote-release#9298]
Type: <b>Bug</b>

According to [this](https://code.visualstudio.com/docs/remote/ssh#_opening-a-terminal-on-a-remote-host), terminals in VSCode run on the remote SSH host, but `code` there should still refer to the local VSCode.
This works great in integrated Command Prompt or Power Shell terminals where executing `code path\to\file` opens a new tab locally with the file.
However, in integrated Git Bash, it tries to open a new VSCode instance on the remote host (it opens as a background process so the GUI never actually shows up and the process just stays in the memory).

This feature is implemented by prepending the `PATH` environment variable in the terminal environment with directory `%USERPROFILE%\.vscode-server\<DEPENDS-ON-VERSION>\bin\remote-cli` that contains `code.cmd` as seen in the output of Command Prompt's `where code`:

```
%USERPOFILE%\.vscode-server\<DEPENDS-ON-VERSION>\bin\remote-cli\code.cmd
%LOCALAPPDATA%\Programs\Microsoft VS Code\bin\code
%LOCALAPPDATA%\Programs\Microsoft VS Code\bin\code.cmd
```

However, that directory is missing the `code` shell script which I believe is causing the problem in Git Bash as it will always start `%LOCALAPPDATA%\Programs\Microsoft VS Code\bin\code` rather than `%USERPOFILE%\.vscode-server\<DEPENDS-ON-VERSION>\bin\remote-cli\code.cmd` as Command Prompt does.

A workaround is to always use `code.cmd` in Git Bash, however, it may be outside the user's control, e.g. https://marketplace.visualstudio.com/items?itemName=eamodio.gitlens runs its own command after initiating rebase.

Extension version: 0.107.1
VS Code version: Code 1.85.0 (af28b32d7e553898b2a91af498b1fb666fdebe0c, 2023-12-06T20:48:09.019Z)
OS version: Windows_NT x64 10.0.19045
Modes:
Remote OS version: Windows_NT x64 10.0.22621

**Comment by joshspicer** [comment]
Thanks for the detailed investigation.  It sounds like the git bash terminal is not getting the updated remote PATH variable applied to it (which i think the server [takes care of here](https://github.com/microsoft/vscode/blob/main/src/vs/server/node/extensionHostConnection.ts#L52)

**Comment by pawelszramowski** [comment]
~~I didn't try `which code` in the integrated Git Bash then so I'm not sure about the `PATH` variable there. It definitely needs to include `%USERPOFILE%\.vscode-server\<DEPENDS-ON-VERSION>\bin\remote-cli` directory, though.~~ `PATH` in the integrated Git Bash includes `%USERPOFILE%\.vscode-server\<DEPENDS-ON-VERSION>\bin\remote-cli`.

Nonetheless, `%USERPOFILE%\.vscode-server\<DEPENDS-ON-VERSION>\bin\remote-cli\code` shell script needs to be provided too (just like `%LOCALAPPDATA%\Programs\Microsoft VS Code\bin` directory has both `code` and `code.cmd` scripts). Otherwise calling `code` in the integrated Git Bash terminal is going to execute `%LOCALAPPDATA%\Programs\Microsoft VS Code\bin\code` instead of `%USERPOFILE%\.vscode-server\<DEPENDS-ON-VERSION>\bin\remote-cli\code`.

**Comment by pawelszramowski** [comment]
@joshspicer I don't agree with your assessment. This is _not_ a duplicate as the other issue regards incorrect `PATH` environment variable.

I ended up checking the variable in the integrated Git Bash during remote connection and it _does_ include the `%USERPOFILE%\.vscode-server\<DEPENDS-ON-VERSION>\bin\remote-cli` directory (as it should).

After I created the following `code` shell script there, the problem was fixed, i.e. executing `code path/to/file` in the integrated Git Bash opened the remote `path/to/file` in my local VSCode rather than a new instance on the remote machine.

```shell
#!/usr/bin/env sh
exec code.cmd "$@"
```

Given how complex `%LOCALAPPDATA%\Programs\Microsoft VS Code\bin\code` is, the `%USERPOFILE%\.vscode-server\<DEPENDS-ON-VERSION>\bin\remote-cli\code` shell script you need to provide has to be more sophisticated than this naive proof of concept.

Please reopen the issue.

**Comment by joshspicer** [comment]
Do you think it's an issue of detecting the wrong platform's command to execute `code` vs `code.cmd`?  Am I correct that that's essentially the workaround you're describing here?

**Comment by pawelszramowski** [comment]
No, this is just how terminals work and is not something VSCode would want to influence.

When you execute `code` in Command Prompt, then it executes the first file with one of the extensions listed in the `PATHEXT` environment variable (e.g. `.COM;.EXE;.BAT;.CMD;.VBS;.VBE;.JS;.JSE;.WSF;.WSH;.MSC;.PY;.PYW`) found in one of the directories listed in the `PATH`. This is `%USERPOFILE%\.vscode-server\<DEPENDS-ON-VERSION>\bin\remote-cli\code.cmd` during remote connection in VSCode.

On the other hand, when you execute `code` in Git Bash then it executes the first executable file found in `PATH` and does not consider `.cmd` executable. Thus, it executes `%LOCALAPPDATA%\Programs\Microsoft VS Code\bin\code` because `%USERPOFILE%\.vscode-server\<DEPENDS-ON-VERSION>\bin\remote-cli\code` shell script is missing.

**Comment by dvirtz** [comment]
I have a similar problem whenever I modify the terminal PATH in settings, e.g.
```
"terminal.integrated.env.linux": {
	"PATH": "${env:PATH}:${env:HOME}/.local/bin:${workspaceFolder:os}/../../../bin",
},
```
